In [2]:
# Importar as bibliotecas necessárias
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

# Carregar os dados
df = pd.read_csv('creditcard.csv')

# Pré-processar os dados
df['Class'] = df['Class'].astype('category')

# Dividir os dados em conjuntos de treinamento e teste
X = df.drop('Amount', axis=1)
y = df['Amount']
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Treinar o modelo
model = LinearRegression()
model.fit(X_train, y_train)

# Avaliar o modelo
y_pred = model.predict(X_test)
mse = mean_squared_error(y_test, y_pred)
print(f'MSE: {mse}')

MSE: 4332.337602906515


O Erro Quadrático Médio (MSE) é uma métrica amplamente utilizada para avaliar a performance de modelos de regressão. Ele calcula a média dos quadrados dos erros entre as previsões do modelo e os valores reais do conjunto de dados de teste. Em outras palavras, o MSE quantifica a qualidade das previsões, medindo o quão distantes estão os pontos previstos do modelo em relação aos pontos reais. Um valor de MSE mais baixo indica uma melhor adequação do modelo aos dados, pois implica em menores discrepâncias entre as previsões e os valores reais. No caso do resultado apresentado, um MSE de 4332.34 sugere que, em média, os quadrados dos erros das previsões estão em torno de 4332.34 unidades ao quadrado, indicando uma certa distância entre as previsões do modelo e os valores reais.